sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml

```diff
@@ -26,6 +26,10 @@
 
   <packaging>jar</packaging>
 
+  <properties>
+    <beam.version>0.4.0-incubating-SNAPSHOT</beam.version>
```

Comment:
Right. This is in the src directory which makes it a template for the generated `pom.xml`. I think it is populated from default values defined in the archetype command the user runs and also this metadata:

https://github.com/apache/incubator-beam/blob/master/sdks/java/maven-archetypes/examples/src/main/resources/META-INF/maven/archetype-metadata.xml

If we could make the arcthetype-metadata.xml use values from the parent-pom to set the defaults, I think that would propagate into the uses of the template.

Agreed it is likely orthogonal.